File: examples/run_monitor.py
# run_monitor.py
# Calls the Page Metadata Extractor Actor (page-metadata-extractor) via the Apify API
# and prints the resulting metadata records once the run finishes.

import os
from apify_client import ApifyClient

# Auth token is read from an env var - never hardcode it.
client = ApifyClient(os.environ["APIFY_TOKEN"])

# Minimal, realistic input matching the Actor's real input schema.
run_input = {
    "startUrls": [{"url": "https://apify.com"}],
    "maxRequestsPerCrawl": 20,
    "onlyChanged": False,
}

print("Starting Page Metadata Extractor run...")

# .call() starts the run on the platform and blocks until it finishes.
run = client.actor("U9fUBHDngX6IyjzzF").call(run_input=run_input)

print(f"Run {run['id']} finished with status: {run['status']}")

# Fetch the extracted metadata records from the run's default dataset.
dataset_items = client.dataset(run["defaultDatasetId"]).list_items().items

print(f"Fetched {len(dataset_items)} metadata records:")
for item in dataset_items:
    print(f"- {item['url']} | \"{item['title']}\" | {item['wordCount']} words | {item['eventType']}")

File: examples/run-monitor.js
// run-monitor.js
// Calls the Page Metadata Extractor Actor (page-metadata-extractor) via the Apify API
// and logs the resulting metadata records once the run finishes.

const { ApifyClient } = require('apify-client');

// Auth token is read from an env var - never hardcode it.
const client = new ApifyClient({
    token: process.env.APIFY_TOKEN,
});

async function main() {
    // Minimal, realistic input matching the Actor's real input schema.
    const input = {
        startUrls: [{ url: 'https://apify.com' }],
        maxRequestsPerCrawl: 20,
        onlyChanged: false,
    };

    console.log('Starting Page Metadata Extractor run...');

    // .call() starts the run on the platform and waits for it to finish.
    const run = await client.actor('U9fUBHDngX6IyjzzF').call(input);

    console.log(`Run ${run.id} finished with status: ${run.status}`);

    // Fetch the extracted metadata records from the run's default dataset.
    const { items } = await client.dataset(run.defaultDatasetId).listItems();

    console.log(`Fetched ${items.length} metadata records:`);
    for (const item of items) {
        console.log(`- ${item.url} | "${item.title}" | ${item.wordCount} words | ${item.eventType}`);
    }
}

main().catch((err) => {
    console.error('Run failed:', err);
    process.exit(1);
});
